testah-junit/src/main/java/org/testah/runner/TestahJUnitRunner.java

```diff
@@ -1,76 +1,90 @@
 package org.testah.runner;
 
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
-
-import org.testah.TS;
-import org.testah.framework.dto.ResultDto;
-import org.testah.runner.testPlan.TestPlanActor;
-
 import akka.actor.ActorRef;
 import akka.actor.ActorSystem;
 import akka.actor.Props;
 import akka.actor.UntypedActor;
 import akka.actor.UntypedActorFactory;
+import org.testah.TS;
+import org.testah.framework.dto.ResultDto;
+import org.testah.framework.testPlan.AbstractTestPlan;
+import org.testah.runner.testPlan.TestPlanActor;
 
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
 /**
  * The Class TestahJUnitRunner.
  */
 public class TestahJUnitRunner {
 
-	/**
-	 * Run tests.
-	 *
-	 * @param numConcurrent the num concurrent
-	 * @param jUnitTestPlanClasses the j unit test plan classes
-	 * @return the list
-	 */
-	public List<ResultDto> runTests(final int numConcurrent, final List<Class<?>> jUnitTestPlanClasses) {
-		if (null != jUnitTestPlanClasses) {
-			return runTests(numConcurrent, new HashSet<Class<?>>(jUnitTestPlanClasses));
-		}
-		return null;
-	}
+    public static boolean inUse = false;
+
+    /**
+     * Run tests.
+     *
+     * @param numConcurrent        the num concurrent
+     * @param jUnitTestPlanClasses the j unit test plan classes
+     * @return the list
+     */
+    public List<ResultDto> runTests(final int numConcurrent, final List<Class<?>> jUnitTestPlanClasses) {
+        setInUse(true);
+        if (null != jUnitTestPlanClasses) {
+            return runTests(numConcurrent, new HashSet<Class<?>>(jUnitTestPlanClasses));
+        }
+        setInUse(false);
+        return null;
+    }
+
+    /**
+     * Run tests.
+     *
+     * @param numConcurrent        the num concurrent
+     * @param jUnitTestPlanClasses the j unit test plan classes
+     * @return the list
+     */
+    public List<ResultDto> runTests(final int numConcurrent, final Set<Class<?>> jUnitTestPlanClasses) {
+        try {
```

Comment:
public List<ResultDto> runTests(final int numConcurrent, final List<Class<?>> jUnitTestPlanClasses) seems to be a wrapper for this method. However, the wrapper also has some logic to deal with list of test plans being null and it has
setInUse(true);
setInUse(false);
Maybe move that logic into this class?